modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -164,6 +181,77 @@ static Mat extractConstant(const Ptr<ImportGraphWrapper>& net, int node_id, int
     }
 }
 
+/*  Slice operator has two optional inputs "axes" and "steps". Some models may be set to have
+    Slice with optional inputs of default values, some of them don't. This Subgraph removes
+    all optional inputs of Slice if values are default.
+*/
+class RemoveSliceAllOptionalInputsSubgraph : public Subgraph {
+ public:
+    RemoveSliceAllOptionalInputsSubgraph(size_t num_inputs = 4) {
+        num_inputs_ = num_inputs;
+
+        int input = addNodeToMatch("");
+        int starts = addNodeToMatch("");
+        int ends = addNodeToMatch("");
+        std::vector<int> inputs{input, starts, ends};
+        for (size_t i = 3; i < num_inputs_; i++) { // axes and steps
+            inputs.push_back(addNodeToMatch(""));
+        }
+
+        slice_id = addNodeToMatch("Slice", inputs);
+
+        setFusedNode("Slice", std::vector<int>{input, starts, ends});
+    }
+
+    virtual bool match(const Ptr<ImportGraphWrapper>& net, int nodeId,
+                       std::vector<int>& matchedNodesIds) CV_OVERRIDE {
+        if (Subgraph::match(net, nodeId, matchedNodesIds)) {
+            if (num_inputs_ >= 4) { // with axes
+                // Check if axes are -1 or last axis
+                auto onnx_net = net.dynamicCast<ONNXGraphWrapper>();
+                int shape_size = onnx_net->getTensorShapeSize(matchedNodesIds[slice_id], 0);
+
+                auto axes = extractConstant(net, matchedNodesIds[slice_id], 3);
+                bool is_default_axes = true;
+                for (size_t i = 0; i < axes.total(); i++) {
+                    const int axis = *(axes.ptr<const int>() + i);
+                    if (axis != -1 && axis != shape_size - 1) {
+                        is_default_axes = false;
+                        break;
+                    }
+                }
+                if (!is_default_axes) {
+                    return false;
+                }
+
+                if (num_inputs_ == 5) {
```

Comment:
There is no common variables required from the `if (num_inputs_ >= 4)` so I recommend to align make two conditions flat:
```
if (num_inputs_ >= 4) {
//...
}
if (num_inputs_ == 5) {
//...
}

Reply:
Done.